### PR TITLE
logs: provide more build info

### PR DESCRIPTION
**Pull request by tbouffard**
Always display the following prior running the build
  - version of the git tool
  - info about the OS
  - commit of this project used to build

Example of generated logs (on windows)
```
OS information
  > Run on Windows
  > Details: Microsoft Windows 10 Pro  64-bit          10.0.18362
  > Generic information: MINGW64_NT-10.0-18362 DESKTOP-TOMA 3.0.7-338.x86_64 2019-07-03 15:16 UTC x86_64 Msys
Build environment
  > Use git version 2.23.0.windows.1
  > Commit: 10d55f61e6d9b80b4b16f169114bfd4be4ecbffb
Build settings
  > BONITA_BPM_VERSION: 7.9.4
  > BONITA_BUILD_NO_CLEAN: false
  > BONITA_BUILD_QUIET: false
  > BONITA_BUILD_STUDIO_ONLY: false
  > BONITA_BUILD_STUDIO_SKIP: false
Prerequisites
  > Use Maven version: 3.6.0
  > Use curl version: 7.65.3
Check if Java version is compatible with Bonita
  > JAVA_HOME is set
  > Java command path is C:\Program Files\Java\jdk8u192-b12/bin/java
  > Java full version: 1.8.0_192
  > Java version: 8
Java version is compatible
```

### CODE REVIEW

**Comment by tbouffard** [comment]
We may also add an issue template for build problems and require users to provide environment information provided thanks to this PR

**Comment by tbouffard** [comment]
✅ on Travis CI build

```
OS information
  > Run on Linux
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.6 LTS"
  > Generic information: Linux travis-job-d52ed476-b158-499a-94ed-123014d097e9 4.15.0-1037-gcp #39~16.04.1-Ubuntu SMP Thu Jul 4 09:29:23 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
Build environment
  > Use git version 2.22.0
  > Commit: 9edb2c4f0d55f113b7da492a95a85b04e78ea27c
Build settings
  > BONITA_BPM_VERSION: 7.9.4
  > BONITA_BUILD_NO_CLEAN: true
  > BONITA_BUILD_QUIET: true
  > BONITA_BUILD_STUDIO_ONLY: false
  > BONITA_BUILD_STUDIO_SKIP: true
Prerequisites
  > Use Maven version: 3.6.0
  > Use curl version: 7.47.0
Check if Java version is compatible with Bonita
  > JAVA_HOME is set
  > Java command path is /usr/lib/jvm/java-1.8.0-openjdk-amd64/bin/java
  > Java full version: 1.8.0_222
  > Java version: 8
Java version is compatible
```

ℹ️ for MacOS info, tested on a playground project using a github action, see https://github.com/tbouffard/github-actions-maven-playground/commit/c765aa24878831a5d3d8cf83eb6a7ddfa3bc00f6/checks?check_suite_id=305958699

```
  > Run on MacOS
ProductName:	Mac OS X
ProductVersion:	10.15.1
BuildVersion:	19B88
  > Generic information: Darwin Mac-994.local 19.0.0 Darwin Kernel Version 19.0.0: Thu Oct 17 16:17:15 PDT 2019; root:xnu-6153.41.3~29/RELEASE_X86_64 x86_6
```

File: build-script.sh [hunk 1]
```diff
@@ -212,28 +212,50 @@ build_gradle_wrapper_test_skip_publishToMavenLocal() {
...
+    if [[ "${OS_IS_LINUX}" == "true" ]]; then
+        echo "  > Run on Linux"
+        echo "$(cat /etc/lsb-release)"
+    elif [[ "${OS_IS_MAC}" == "true" ]]; then
+        echo "  > Run on MacOS"
+        echo "$(sw_vers)"
+    elif [[ "${OS_IS_WINDOWS}" == "true" ]]; then
+        echo "  > Run on Windows"
+        echo "  > Details: $(wmic os get Caption,Version,osarchitecture | grep Microsoft)"
+    else
```
Comment (amottier) [inline build-script.sh]: I think it is not possible to follow into this else condition as OS_IS_LINUX will be by default set to true.

Comment (tbouffard) [reply build-script.sh]: Yes, I will drop this condition

**Comment by amottier** [comment]
LGTM. @tbouffard do you plan additional changes or is the PR ready to be merged?

**Comment by tbouffard** [comment]
@amottier thanks
I have added the 2 last commit to improve the logs and I am done with that
So feel free to merge

About an issue template, we can manage to add it in a dedicated PR.

**Comment by tbouffard** [comment]
Final implementation produces logs like in the following

**on Ubuntu, run by Travis CI**
```
OS information
  > Run on Linux
      DISTRIB_ID=Ubuntu
      DISTRIB_RELEASE=16.04
      DISTRIB_CODENAME=xenial
      DISTRIB_DESCRIPTION=Ubuntu 16.04.6 LTS
  > Generic information: Linux travis-job-1c930c75-7b9d-4e6f-8dda-6bc2a0a8bc08 4.15.0-1037-gcp #39~16.04.1-Ubuntu SMP Thu Jul 4 09:29:23 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
Build environment
  > Use git version 2.22.0
  > Commit: 7088ad87f7b162db8a4f6bfd6490821b83caca19
Build settings
  > BONITA_BPM_VERSION: 7.9.4
  > BONITA_BUILD_NO_CLEAN: true
  > BONITA_BUILD_QUIET: true
  > BONITA_BUILD_STUDIO_ONLY: false
  > BONITA_BUILD_STUDIO_SKIP: true
Prerequisites
  > Use Maven version: 3.6.0
  > Use curl version: 7.47.0
  > Java prerequisites
      Check if Java version is compatible with Bonita
      JAVA_HOME is set
      Java command path is /usr/lib/jvm/java-1.8.0-openjdk-amd64/bin/java
      JVM details
        openjdk version 1.8.0_222
        OpenJDK Runtime Environment (build 1.8.0_222-8u222-b10-1ubuntu1~16.04.1-b10)
        OpenJDK 64-Bit Server VM (build 25.222-b10, mixed mode)
      Java full version: 1.8.0_222
      Java version: 8
      Java version is compatible with Bonita
```

**on MacOS, run by Github Actions**
```
      BuildVersion: 19B88
  > Generic information: Darwin Mac-1131.local 19.0.0 Darwin Kernel Version 19.0.0: Thu Oct 17 16:17:15 PDT 2019; root:xnu-6153.41.3~29/RELEASE_X86_64 x86_64
Build environment
  > Use git version 2.23.0
  > Commit: 4c054d6c370b1fb7372cf974505670ac5a2bb78a
Build settings
  > BONITA_BPM_VERSION: 7.9.4
  > BONITA_BUILD_NO_CLEAN: false
  > BONITA_BUILD_QUIET: false
  > BONITA_BUILD_STUDIO_ONLY: false
  > BONITA_BUILD_STUDIO_SKIP: false
Prerequisites
  > Use Maven version: 3.6.2
  > Use curl version: 7.66.0
  > Java prerequisites
      Check if Java version is compatible with Bonita
      JAVA_HOME is set
      Java command path is /Library/Java/JavaVirtualMachines/zulu-8.jdk/Contents/Home/bin/java
      JVM details
        openjdk version 1.8.0_232
        OpenJDK Runtime Environment (Zulu 8.42.0.21-CA-macosx) (build 1.8.0_232-b18)
        OpenJDK 64-Bit Server VM (Zulu 8.42.0.21-CA-macosx) (build 25.232-b18, mixed mode)
      Java full version: 1.8.0_232
      Java version: 8
      Java version is compatible with Bonita
```

**on Windows, run by Github Actions**
```
OS information
  > Run on Windows
      Caption=Microsoft Windows Server 2019 Datacenter
      OSArchitecture=64-bit
      Version=10.0.17763
  > Generic information: MINGW64_NT-10.0-17763 fv-az75 3.0.7-338.x86_64 2019-07-03 15:16 UTC x86_64 Msys
Build environment
  > Use git version 2.23.0.windows.1
  > Commit: 4c054d6c370b1fb7372cf974505670ac5a2bb78a
Build settings
  > BONITA_BPM_VERSION: 7.9.4
  > BONITA_BUILD_NO_CLEAN: false
  > BONITA_BUILD_QUIET: false
  > BONITA_BUILD_STUDIO_ONLY: false
  > BONITA_BUILD_STUDIO_SKIP: false
Prerequisites
  > Use Maven version: 3.6.2
  > Use curl version: 7.65.3
  > Java prerequisites
      Check if Java version is compatible with Bonita
      JAVA_HOME is set
      Java command path is C:\Program Files\Java\zulu-8-azure-jdk_8.40.0.25-8.0.222-win_x64/bin/java
      JVM details
        openjdk version 1.8.0_222
        OpenJDK Runtime Environment (Zulu 8.40.0.25-win64)-Microsoft-Azure-restricted (build 1.8.0_222-b10)
        OpenJDK 64-Bit Server VM (Zulu 8.40.0.25-win64)-Microsoft-Azure-restricted (build 25.222-b10, mixed mode)
      Java full version: 1.8.0_222
      Java version: 8
      Java version is compatible with Bonita```

**Pull request merged**